### ISSUE TITLE
About different patch embedding in ViT

### ISSUE THREAD

**Issue by wddwzwhhxx** [lucidrains/vit-pytorch#184]
I saw two versions of patch embedding：
1. "patch-embedding" in vit-pytorch
self.to_patch_embedding = nn.Sequential(
            Rearrange('b c (h p1) (w p2) -> b (h w) (p1 p2 c)', p1 = patch_height, p2 = patch_width),
            nn.Linear(patch_dim, dim),
        )
2.  "patch-embedding" in timm
self.proj = nn.Conv2d(in_chans, embed_dim, kernel_size=patch_size, stride=patch_size)
x = self.proj(x).flatten(2).transpose(1, 2)

Is there any performance difference between the two "patch-embedding"?
thanks!

**Comment by lucidrains** [comment]
@wddwzwhhxx would highly recommend timm instead of vit-pytorch if you care about performance

my repositories usually optimize for clarity!
